### PR TITLE
use strict comparision

### PR DESCRIPTION
**Pull request by keradus**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | kind of
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | see CI service
| Fixed tickets | 
| License       | MIT
| Doc PR        | n/a

ref https://github.com/symfony/symfony/pull/24084#issuecomment-326972535

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
How many of these will lead to BC breaks?

**Comment by keradus** [comment]
with good code, none ;)
I would actually treat it as bugfix, for example, if we expected a number, but someone provided numeric-string, currently handled it for him, now we won't, as in the end he provided wrong type. nice in context of moving to php-7 strict types

**Comment by fabpot** [comment]
I'm strongly against such a change (which was rejected more than once before). It does not fix anything and will lead to BC breaks. 👎 from me.

**Comment by keradus** [comment]
we shall know the code, and if we do, we do know the types. for that, no need for type juggling ;)
I agree that it will prevent end-user to provide not type-compatible value to pass now.
What about a change like this for major release?

**Comment by nicolas-grekas** [comment]
:-1: also for the same reasons. Reason about code in the abstract and enforcing policies is fine for future code. But this is existing code that works really fine, this won't improve anything.
:-1: also for applying on master, that's again not improving anything and will just makes our lives as mergers more difficult.

File: src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php [hunk 1]
```diff
@@ -83,7 +83,7 @@ public function mapFormsToData($forms, &$data)
             if (null !== $propertyPath && $config->getMapped() && $form->isSubmitted() && $form->isSynchronized() && !$form->isDisabled()) {
                 // If the field is of type DateTime and the data is the same skip the update to
                 // keep the original object hash
-                if ($form->getData() instanceof \DateTime && $form->getData() == $this->propertyAccessor->getValue($data, $propertyPath)) {
+                if ($form->getData() instanceof \DateTime && $form->getData() === $this->propertyAccessor->getValue($data, $propertyPath)) {
```
Comment (gharlan) [inline src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: @keradus This is an example, where strict comparison breaks the code: https://3v4l.org/TqWYL

Comment (keradus) [reply src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: nah, this is example where comparision shall be done other way (eg, comparing timestamps)

**Pull request closed without merging**